client/src/main/scala/com/aliyun/emr/rss/client/write/LifecycleManager.scala

```diff
@@ -1144,12 +1231,13 @@ class LifecycleManager(appId: String, val conf: RssConf) extends RpcEndpoint wit
     masterAndSlavePairs._1.add(masterLocation)
   }
 
-  private def reallocateForNonExistPartitionLocationFromCandidates(
-      partitionId: Int,
-      oldEpochId: Int,
+  private def reallocateChangePartitionRequestSlotsFromCandidates(
+      oldPartitions: List[ChangePartitionRequest],
       candidates: List[WorkerInfo]): WorkerResource = {
     val slots = new WorkerResource()
-    allocateFromCandidates(partitionId, oldEpochId, candidates, slots)
+    oldPartitions.foreach { partition =>
```

Comment:
There are cases where oldPartition is null, see ShuffleClientImpl.pushOrMergeData
```angular2html
    // get location
    if (!map.containsKey(partitionId)
        && !revive(
            applicationId,
            shuffleId,
            mapId,
            attemptId,
            partitionId,
            0,
            null,
            StatusCode.PUSH_DATA_FAIL_NON_CRITICAL_CAUSE)) {
      throw new IOException(
          "Revive for shuffle " + shuffleKey + " partitionId " + partitionId + " failed.");
    }
```

Reply:
> There are cases where oldPartition is null, see ShuffleClientImpl.pushOrMergeData
> 
> ```
>     // get location
>     if (!map.containsKey(partitionId)
>         && !revive(
>             applicationId,
>             shuffleId,
>             mapId,
>             attemptId,
>             partitionId,
>             0,
>             null,
>             StatusCode.PUSH_DATA_FAIL_NON_CRITICAL_CAUSE)) {
>       throw new IOException(
>           "Revive for shuffle " + shuffleKey + " partitionId " + partitionId + " failed.");
>     }
> ```

A mistake, here I forgot to change parameter name, should be `changePartitionRequests`